### PR TITLE
Standardize refarch and demos / AKA the MarkDembo show

### PR DESCRIPTION
**Pull request by kodster28**
### Summary

Following the pattern introduced in #15894 (and the KV work that's happening in #15884), adding a new section to `/demos` pages that includes related reference architectures.

Open to discussing better organization / treatments in the future... but think this gets these resources in the hands of folks who'd be interested.

cc: @markdembo, it's really just highlighting all the hard work you've done (and hopefully continue to do).

### CODE REVIEW

**Comment by kodster28** [comment]
> This PR changes current filenames or deletes current files. Make sure you have [redirects](https://developers.cloudflare.com/pages/configuration/redirects/) set up to cover the following paths:
> 
> * [x]  `/ai-gateway/architectures/`

This doesn't require a redirect b/c it's just a placeholder navigation link

**Comment by thomasgauvin** [comment]
I like this. I have received a couple asks from customers to see full-blown use cases/architectures and I think this is a good way to link back to reference architectures center and make these discoverable from product docs themselves. This way, we have, in increasing complexity order, API -> Examples -> Tutorial -> Demos & Architectures, to give customers the full story and help them see how others are using these services.

File: content/d1/demos.md [hunk 1]
```diff
@@ -1,11 +1,21 @@
 ---
 pcx_content_type: navigation
-title: Demos
+title: Demos and architectures
 weight: 8
 ---
 
-# Demo apps
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/d1/demos.md]:
```suggestion
# Demo applications and reference architectures
```

File: content/durable-objects/demos.md [hunk 2]
```diff
@@ -1,11 +1,21 @@
 ---
 pcx_content_type: navigation
-title: Demos
+title: Demos and architectures
 weight: 8
 ---
 
-# Demo apps
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/durable-objects/demos.md]:
```suggestion
# Demo applications and reference architectures
```

File: content/hyperdrive/demos.md [hunk 3]
```diff
@@ -1,11 +1,21 @@
 ---
 pcx_content_type: navigation
-title: Demos
+title: Demos and architectures
 weight: 7
 ---
 
-# Demo apps
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/hyperdrive/demos.md]:
```suggestion
# Demo applications and reference architectures
```

File: content/pages/demos.md [hunk 4]
```diff
@@ -1,11 +1,21 @@
 ---
 pcx_content_type: navigation
-title: Demos
+title: Demos and architectures
 weight: 8
 ---
 
-# Demo apps
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/pages/demos.md]:
```suggestion
# Demo applications and reference architectures
```

File: content/queues/demos.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+---
+pcx_content_type: navigation
+title: Demos and architectures
+weight: 7
+---
+
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/queues/demos.md]:
```suggestion
# Demo applications and reference architectures
```

File: content/r2/demos.md [hunk 6]
```diff
@@ -1,11 +1,21 @@
 ---
 pcx_content_type: navigation
-title: Demos
+title: Demos and architectures
 weight: 9
 ---
 
-# Demo apps
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/r2/demos.md]:
```suggestion
# Demo applications and reference architectures
```

File: content/turnstile/demos.md [hunk 7]
```diff
@@ -1,11 +1,21 @@
 ---
 pcx_content_type: navigation
-title: Demos
-weight: 5
+title: Demos and architectures
+weight: 6
 ---
 
-# Demo apps
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/turnstile/demos.md]:
```suggestion
# Demo applications and reference architectures
```

File: content/workers/demos.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
+---
+pcx_content_type: navigation
+title: Demos and architectures
+weight: 5
+---
+
+# Demo Applications and Reference Architectures
```
Comment (marciocloudflare) [inline content/workers/demos.md]:
```suggestion
# Demo applications and reference architectures
```

**Review by marciocloudflare — APPROVED** [review]

**Pull request merged**